app/api/__tests__/get-random-whys.js

```diff
@@ -0,0 +1,242 @@
+
+  const user = { id: USER1 }
+  const cb = jest.fn()
+
+  await getRandomWhys.call({ synuser: user }, POINT1.toString(), 'most', 5, cb)
+
+  expect(cb).toHaveBeenCalledTimes(1)
+  const result = cb.mock.calls[0][0]
+//   console.log('Test result:', result)
+  expect(result.length).toBe(5)
+  result.forEach(why => {
+    expect(why).not.toHaveProperty('userId')
+    expect(why.parentId.toString()).toBe(POINT1.toString())
+    expect(why.category).toBe('most')
+  })
+})
+
+test('Case where there are 10 matching whys and qty is 5', async () => {
+  const whys = [
+    {
+      _id: new ObjectId(),
+      title: 'Why1',
+      description: 'Description 1',
+      round: 1,
+      parentId: POINT1,
+      userId: USER1,
+      category: 'most',
+    },
+    {
+      _id: new ObjectId(),
+      title: 'Why2',
+      description: 'Description 2',
+      round: 1,
+      parentId: POINT1,
+      userId: USER1,
+      category: 'most',
+    },
+    {
+      _id: new ObjectId(),
+      title: 'Why3',
+      description: 'Description 3',
+      round: 1,
+      parentId: POINT1,
+      userId: USER1,
+      category: 'most',
+    },
+    {
+      _id: new ObjectId(),
+      title: 'Why4',
+      description: 'Description 4',
+      round: 1,
+      parentId: POINT1,
+      userId: USER1,
+      category: 'most',
+    },
+    {
+      _id: new ObjectId(),
+      title: 'Why5',
+      description: 'Description 5',
+      round: 1,
+      parentId: POINT1,
+      userId: USER1,
+      category: 'most',
+    },
+    {
+      _id: new ObjectId(),
+      title: 'Why6',
+      description: 'Description 6',
+      round: 1,
+      parentId: POINT1,
+      userId: USER1,
+      category: 'most',
+    },
+    {
+      _id: new ObjectId(),
+      title: 'Why7',
+      description: 'Description 7',
+      round: 1,
+      parentId: POINT1,
+      userId: USER1,
+      category: 'most',
+    },
+    {
+      _id: new ObjectId(),
+      title: 'Why8',
+      description: 'Description 8',
+      round: 1,
+      parentId: POINT1,
+      userId: USER1,
+      category: 'most',
+    },
+    {
+      _id: new ObjectId(),
+      title: 'Why9',
+      description: 'Description 9',
+      round: 1,
+      parentId: POINT1,
+      userId: USER1,
+      category: 'most',
+    },
+    {
+      _id: new ObjectId(),
+      title: 'Why10',
+      description: 'Description 10',
+      round: 1,
+      parentId: POINT1,
+      userId: USER1,
+      category: 'most',
+    },
+  ]
+  await Mongo.db.collection('points').insertMany(whys)
+
+  const user = { id: USER1 }
+  const cb = jest.fn()
+
+  await getRandomWhys.call({ synuser: user }, POINT1.toString(), 'most', 5, cb)
+
+  expect(cb).toHaveBeenCalledTimes(1)
+  const result = cb.mock.calls[0][0]
+  expect(result.length).toBe(5)
+  result.forEach(why => {
+    expect(why).not.toHaveProperty('userId')
+    expect(why.parentId.toString()).toBe(POINT1.toString())
+    expect(why.category).toBe('most')
+  })
```

Comment:
I'm trying to figure out a test for randomness.  Here's an idea.  Call getRandomWhy's once, and then call it again and compare the two lists and make sure they are not the same.